dali/pipeline/operators/reader/coco_reader_op_test.cc

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "dali/test/dali_test_config.h"
+#include "dali/pipeline/pipeline.h"
+
+namespace dali {
+
+class CocoReaderTest : public ::testing::Test {
+ public:
+  std::vector<std::pair<std::string, std::string>> Outputs() {
+    return {{"images", "cpu"}, {"boxes", "cpu"}, {"labels", "cpu"}, {"image_ids", "cpu"}};
+  }
+
+  OpSpec CocoReaderOpSpec() {
+    return OpSpec("COCOReader")
+          .AddArg("device", "cpu")
+          .AddArg("file_root", file_root_)
+          .AddArg("annotations_file", annotations_filename_)
+          .AddArg("save_img_ids", true)
+          .AddOutput("images", "cpu")
+          .AddOutput("boxes", "cpu")
+          .AddOutput("labels", "cpu")
+          .AddOutput("image_ids", "cpu");
+  }
+
+  int SmallCocoSize() { return 64; }
+  int EmptyImages() { return 2; }
+  int ImagesWithBigObjects() { return 2; }
+  int ObjectCount() { return 194; }
+
+  std::vector<int> CopyIds(DeviceWorkspace &ws) {
+    auto &output = ws.Output<dali::CPUBackend>(3);
+    auto shape = output.shape();
+
+    vector<int> ids(shape.size());
+
+    MemCopy(
+      ids.data(),
+      output.data<int>(),
+      shape.size() * sizeof(int));
```

Comment:
I'm not 100% sure here, but `sizeof(int)` seems to be unsafe, cause it's size depends on architecture. Might this be a problem?

Reply:
Should be fine, since the data type is also `int`